### PR TITLE
feat: DCMAW-10311 generate key pair

### PR DESCRIPTION
**Pull request by alex-bradbury-gds**
### Description of changes
- moving current login related code into separate package
- moving app checker functionality into authentication repo from one-login
- create ClientAttestationManager interface
- a Firebase implementation of the AttestationManager is created with 'empty' methods
- `KeystoreManager` created to create a way to interact with the Keystore and use keys from it. Currently it generates an EC keypair on initialization if one does not already exist

Evidence against the ACs will be found in the One-Login codebase although even there it will be difficult to really show the keys being created. Hopefully the tests present here are covering the desired behaviour.

[DCMAW-10311]: https://govukverify.atlassian.net/browse/DCMAW-10311?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by BiancaMihaila — APPROVED** [review]

File: app/build.gradle.kts [hunk 1]
```diff
@@ -101,6 +101,7 @@ dependencies {
         libs.appcompat,
         libs.appauth,
         libs.kotlinx.serialization.json,
+        libs.network,
```
Comment (obinns-dd) [inline app/build.gradle.kts:104]: I'm quite nervous about importing core modules into other core modules, it feels like a violation of dependency inversion principle. Let's have a chat on Monday 🙂

Comment (ThomasIent) [reply app/build.gradle.kts]: Network module has been removed along with the build logic to include that project.

File: settings.gradle.kts [hunk 2]
```diff
@@ -13,6 +15,37 @@ dependencyResolutionManagement {
...
+        gprUser.get() to gprToken.get()
+    } catch (exception: MissingValueException) {
+        logger.warn(
+            "Could not find 'Github Package Registry' properties. Refer to the proceeding " +
+                    "location for instructions:\n\n" +
+                    "${rootDir.path}/docs/developerSetup/github-authentication.md\n",
+            exception
+        )
+
+        System.getenv("USERNAME") to System.getenv("TOKEN")
```
Comment (BiancaMihaila) [inline settings.gradle.kts:48]: Do we still need all of this since not using the networking package here?

Comment (ThomasIent) [reply settings.gradle.kts:48]: No, good spot. I thought I'd got rid of all the build stuff

**Pull request merged**